worker/task/process_broadcast_event_creation.go

```diff
@@ -0,0 +1,129 @@
+package task
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/frain-dev/convoy/util"
+
+	"github.com/frain-dev/convoy/pkg/msgpack"
+
+	"github.com/frain-dev/convoy/api/models"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/pkg/log"
+	"github.com/frain-dev/convoy/queue"
+	"github.com/hibiken/asynq"
+	"github.com/oklog/ulid/v2"
+)
+
+func ProcessBroadcastEventCreation(endpointRepo datastore.EndpointRepository, eventRepo datastore.EventRepository, projectRepo datastore.ProjectRepository, eventDeliveryRepo datastore.EventDeliveryRepository, eventQueue queue.Queuer, subRepo datastore.SubscriptionRepository, deviceRepo datastore.DeviceRepository) func(context.Context, *asynq.Task) error {
+	return func(ctx context.Context, t *asynq.Task) error {
+		var broadcastEvent models.BroadcastEvent
+
+		err := msgpack.DecodeMsgPack(t.Payload(), &broadcastEvent)
+		if err != nil {
+			err := json.Unmarshal(t.Payload(), &broadcastEvent)
+			if err != nil {
+				return &EndpointError{Err: err, delay: defaultDelay}
+			}
```

Comment:
Do we need this since we're sure it will be msgpack?

Reply:
@jirevwe how do you mean we're sure?

Reply:
@jirevwe thanks, fixed!